### PR TITLE
:seedling: Consultancy Hero Section Component #176

### LINKED ISSUE #176: 🎨 Create the consultancy hero section
#  Create the consultancy hero section 
---

Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)

---

### Summary

As a user, I want a nice hero section on the consultancy page that captures my attention and introduces the page content. Reuse the text-content component created in #20, and the reusable slider button in  #24. I want a good experience on the web and phone.

Design:

consultancy hero section

Link to full design:

```
https://xd.adobe.com/view/b98caf27-6549-49e5-8ed8-c5234788df61-ecdf/screen/cc35e19a-24f2-42ce-8803-ee2942839528/
```
 
### Technical Analysis

This feature can be found in libs/features/pages/consultancy-page inside a components folder. The component's name is consultancy-hero-section.

### Acceptance Criteria

1. When rendered the component is pixel perfect compared to the design above.
2. The component is clean and does not have more than 30 lines
3. The component SCSS uses CSS variables for colour, font, ... Connected to the global style.scss
5. The design is responsive for screen, tablet and phone

### PR DESCRIPTION
**Pull request by preston-56**
# Description

- Created Hero section on the consultancy page that introduces the page content.
- Reused the text-content component created in https://github.com/italanta/elewa-website/issues/20, and the reusable slider button in https://github.com/italanta/elewa-website/issues/24.

Fixes #176 

## Screenshot
Screenshot from 2023-09-05 17-42-43

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.scss [hunk 1]
```diff
@@ -0,0 +1,89 @@
+/* Layout Properties for larger screens */
+.consultancy-main {
+    top: 0px;
+    left: 0px;
+    width: 1920px;
+    height: 1082px;
+    /**UI Properties*/
+    background: #dbc2f5 0% 0% no-repeat padding-box;
```
Comment (AnnaL001) [inline libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.scss:8]: For reuse of colors, use CSS variables specified in the elements/theming/_variables.scss file to specify colors/background colors

File: libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.scss [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    margin-right: auto;
+    max-width: 1920px;
+  }
+  
+  .text-section {
+    left: 209px;
+    width: 839px;
+    height: 106px;
+    text-align: left;
+    font: normal normal medium 82px/110px DM Sans;
```
Comment (AnnaL001) [inline libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.scss:20]: Also here, for the font, use CSS variables specified in the elements/theming/_variables.scss file

File: libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.scss [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    letter-spacing: 0px;
+    color: #000000;
```
Comment (AnnaL001) [inline libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.scss:22]: Same here

Comment (AnnaL001) [inline libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.scss:22]: Same here

File: libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.scss [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  
+  /* Styling for the second circle */
+  .circle.circle-2 {
+    top: 229px;
+    left: 846px;
+    width: 1065px;
+    height: 1056px;
+    transform: matrix(-1, 0, 0, -1, 0, 0);
+    background: transparent linear-gradient(90deg, #dbc2f500 0%, #dbc2f5e6 100%)
+      0% 0% no-repeat padding-box;
```
Comment (AnnaL001) [inline libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.scss:63]: Same here

File: libs/features/pages/consultancy-page/src/lib/consultancy-page.module.ts [hunk 5]
```diff
@@ -4,9 +4,19 @@ import { CommonModule } from '@angular/common';
 import { ConsultancyRoutingModule } from './consultancy.routing';
 
 import { ConsultancyPageComponent } from './main/consultancy-page/consultancy-page.component';
+import { ConsultancyHeroSectionComponent } from './consultancy-hero-section/consultancy-hero-section.component';
+import { TextsModule } from '@elewa-website/elements/layout/texts';
+import { AppHeaderModule } from '@elewa-website/elements/layout/header';
+import { ButtonsModule } from '@elewa-website/elements/layout/buttons';
```
Comment (AnnaL001) [inline libs/features/pages/consultancy-page/src/lib/consultancy-page.module.ts:11]: Check the order of imports; elements imports should appear before local imports

File: libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.scss [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  }
+  
+  /* Styling for the first circle */
+  .circle.circle-1 {
+    top: 229px;
+    left: 219px;
+    width: 1065px;
+    height: 1056px;
+    background: transparent linear-gradient(90deg, #dbc2f500 0%, #dbc2f5e6 100%)
+      0% 0% no-repeat padding-box;
```
[the remark this answers is not in the document]
Comment (preston-56) [reply libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.scss]: @AnnaL001 *resolved

File: libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.scss (hunk 3 above)
Comment (preston-56) [reply libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.scss]: @AnnaL001 *resolved

File: libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (preston-56) [reply libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.scss]: @AnnaL001 *resolved

File: libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (preston-56) [reply libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.scss]: @AnnaL001 *resolved

File: libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.scss (hunk 4 above)
Comment (preston-56) [reply libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.scss]: @AnnaL001 *resolved

File: libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  styleUrls: ['./consultancy-hero-section.component.scss'],
+})
+export class ConsultancyHeroSectionComponent {
+  heroTextData = {
+    title: 'Learning design',
+    description: 'Your organisation has the potential to turn every task into an opportunity to learn. From growing crops, to nursing someone to good health, to developing the next-generation technology. Let’s work together to make the magic happen.',
+  };
+  sliderButtonData = {
+    text: 'Unlock your potential',
+  };
```
[the remark this answers is not in the document]
Comment (preston-56) [reply libs/features/pages/consultancy-page/src/lib/consultancy-hero-section/consultancy-hero-section.component.ts]: @AnnaL001 *resolved

File: libs/features/pages/consultancy-page/src/lib/consultancy-page.module.ts (hunk 5 above)
Comment (preston-56) [reply libs/features/pages/consultancy-page/src/lib/consultancy-page.module.ts]: @AnnaL001 *resolved

Thanks again!

**Comment by LemmyMwaura** [comment]
Very Good job on this but it's already been implemented.

**Pull request closed without merging**